### PR TITLE
[CARBONDATA-899] Added Support for DecimalType and Timestamp for spark-2.1

### PR DESCRIPTION
**Pull request by bhavya411**
Added Support for DecimalType and Timestamp with Spark-2.1
Local Build is running fine.
Manually Tested the functionality on Presto CLI

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1566/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1585/

**Comment by chenliang613** [comment]
@bhavya411  verified, it works fine.
Can you change title to : Added Support for DecimalType and Timestamp for spark-2.1 for integration/presto,  please change JIRA description accordingly.

File: integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+public class CarbondataPageSource implements ConnectorPageSource {
+
+  private static final int ROWS_PER_REQUEST = 4096;
+  private final RecordCursor cursor;
+  private final List<Type> types;
+  private final PageBuilder pageBuilder;
+  private boolean closed;
+  private final char[] buffer = new char[100];
+
+
```
Comment (chenliang613) [inline integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java]: please remove the redundant blank.  same for other place also.

File: integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+import static com.facebook.presto.spi.type.Decimals.encodeUnscaledValue;
+import static com.facebook.presto.spi.type.Decimals.isShortDecimal;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.math.RoundingMode.HALF_UP;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+
+public class CarbondataPageSource implements ConnectorPageSource {
```
Comment (chenliang613) [inline integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java]: please provide comment for the new class of CarbondataPageSource

**Comment by bhavya411** [comment]
I have fixed the spacing issues as well as indentations , please review.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1601/

**Comment by chenliang613** [comment]
LGTM

**Comment by chenliang613** [comment]
@bhavya411  can you squash all commits into one commit.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1658/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1659/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1660/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1661/

**Pull request merged**